src/allocators/HybridAllocator.sol

```diff
@@ -82,61 +86,59 @@ contract HybridAllocator is IHybridAllocator {
             ALLOCATOR_ID = allocatorId;
         }
 
-        signers[signer_] = true;
-        signerCount++;
+        owner = owner_;
+        addSigner(signer_);
 
-        emit AllocatorInitialized(AL.THE_COMPACT, signer_, ALLOCATOR_ID);
-        emit SignerAdded(signer_);
+        emit AllocatorInitialized(AL.THE_COMPACT, owner_, ALLOCATOR_ID);
     }
 
     /// @inheritdoc IHybridAllocator
-    function addSigner(address signer_) external onlySigner {
+    function addSigner(address signer_) public onlyOwner {
         if (signer_ == address(0) || signers[signer_]) {
             revert InvalidSigner();
         }
         signers[signer_] = true;
-        signerCount++;
         emit SignerAdded(signer_);
     }
 
     /// @inheritdoc IHybridAllocator
-    function removeSigner(address signer_) external onlySigner {
-        if (signerCount == 1) {
-            revert LastSigner();
-        }
+    function removeSigner(address signer_) public onlyOwner {
         if (!signers[signer_]) {
             revert InvalidSigner();
         }
-        // Clear any pending replacement proposed by this signer
-        delete pendingSignerReplacement[signer_];
 
         signers[signer_] = false;
-        signerCount--;
         emit SignerRemoved(signer_);
     }
 
     /// @inheritdoc IHybridAllocator
-    function replaceSigner(address newSigner_) external onlySigner {
-        if (newSigner_ == address(0) || signers[newSigner_]) {
+    function replaceSigner(address oldSigner_, address newSigner_) external onlyOwner {
+        if (oldSigner_ == newSigner_) {
             revert InvalidSigner();
         }
-        address oldSigner = msg.sender;
-        pendingSignerReplacement[oldSigner] = newSigner_;
-        emit SignerReplacementProposed(oldSigner, newSigner_);
+        removeSigner(oldSigner_);
+        addSigner(newSigner_);
     }
 
-    function acceptSignerReplacement(address oldSigner_) external {
-        address newSigner_ = pendingSignerReplacement[oldSigner_];
-        if (newSigner_ == address(0) || msg.sender != newSigner_) {
-            revert InvalidSigner();
+    /// @inheritdoc IHybridAllocator
+    function proposeOwnerReplacement(address newOwner_) external onlyOwner {
```

Comment:
i actually think we should consider allowing address(0) here so you can "cancel" an ownership transfer; erroneously putting in address(0) is not that big of a deal since nothing will happen